ReactiveCocoa/DelegateProxy.swift

```diff
@@ -70,9 +70,9 @@ extension DelegateProxy {
 				return proxy
 			}
 
-			let newSetterImpl: @convention(block) (NSObject, NSObject) -> Void = { object, delegate in
+			let newSetterImpl: @convention(block) (NSObject, AnyObject?) -> Void = { object, delegate in
 				let proxy = object.associations.value(forKey: key)!
-				proxy.forwardee = (delegate as! Delegate)
+				proxy.forwardee = (delegate as! Delegate?)
```

Comment:
This is the culprit. @akashivskyy reported a EXC_BAD_ACCESS trap in the `[UISearchController dealloc]`, which uncovered the mishandling of an optional argument here.